driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -36,14 +37,29 @@ impl LinearOptimisationPriceFinder {
     }
 }
 
-fn token_id(token: usize) -> String {
+fn token_id(token: u16) -> String {
     format!("token{}", token)
 }
 
 fn account_id(account: H160) -> String {
     format!("{:x}", account)
 }
 
+fn serialize_tokens(orders: &[models::Order]) -> Vec<String> {
+    // Get collection of all token ids appearing in orders
+    let mut token_ids = orders.iter().map(|o| o.buy_token).collect::<Vec<u16>>();
```

Comment:
nit: why not use a HashSet to start with? HashSets should also implement the Extend trait (https://doc.rust-lang.org/src/std/collections/hash/set.rs.html#938-946)

Reply:
Believe me I tried (for days)... but I will try again.

Reply:
Very nice. Worked like a charm!